tests/rptest/tests/schema_registry_test.py

```diff
@@ -1604,3 +1605,40 @@ def test_protobuf(self):
         self.logger.info(result_raw)
         assert result_raw.status_code == requests.codes.ok
         assert result_raw.json() == [2]
+
+
+class SchemaRegistryAutoAuthTest(SchemaRegistryEndpoints):
+    """
+    Test schema registry against a redpanda cluster with Auto Auth enabled.
+    """
+    def __init__(self, context):
+        security = SecurityConfig()
+        security.kafka_enable_authorization = True
+        security.endpoint_authn_method = 'sasl'
+        security.auto_auth = True
+
+        super(SchemaRegistryAutoAuthTest, self).__init__(context,
+                                                         security=security)
+
+    @cluster(num_nodes=3)
+    def test_get_subjects(self):
+        admin = Admin(self.redpanda)
+
+        def check_connection(hostname: str):
+            result_raw = self._get_subjects(hostname=hostname)
+            self.logger.info(result_raw.status_code)
+            self.logger.info(result_raw.json())
+            assert result_raw.status_code == requests.codes.ok
+            assert result_raw.json() == []
```

Comment:
1. it'd be nice if either the negative side of this could be tested (e.g. turn off auto-auth in client and the test otherwise unchanged will fail) or if we could see a metric or other metadata that implies that auth is being used.
2. can all the other test suites run now with auto auth on/off?

Reply:
1. Good call
2. I would think so